components/usage/pkg/apiv1/usage.go

```diff
@@ -453,3 +454,48 @@ func NewUsageService(conn *gorm.DB, pricer *WorkspacePricer, costCenterManager *
 		pricer: pricer,
 	}
 }
+
+func (s *UsageService) AddUsageCreditNote(ctx context.Context, req *v1.AddUsageCreditNoteRequest) (*v1.AddUsageCreditNoteResponse, error) {
+	log.Log.
+		WithField("attribution_id", req.AttributionId).
+		WithField("credits", req.Credits).
+		WithField("user", req.UserId).
+		WithField("note", req.Note).
+		Info("Adding usage credit note.")
+
+	attributionId, err := db.ParseAttributionID(req.AttributionId)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "AttributionID '%s' couldn't be parsed (error: %s).", req.AttributionId, err)
+	}
+
+	note := strings.TrimSpace(req.Note)
+	if note == "" {
+		return nil, status.Error(codes.InvalidArgument, "The note must not be empty.")
+	}
+
+	userId, err := uuid.Parse(req.UserId)
+	if err != nil {
+		return nil, fmt.Errorf("The user id is not a valid UUID. %w", err)
+	}
+
+	usage := db.Usage{
+		ID:            uuid.New(),
+		AttributionID: attributionId,
+		Description:   note,
+		CreditCents:   db.NewCreditCents(float64(req.Credits * -1)),
```

Comment:
:cloud: Would it make sense to check for the sign of `req.Credits`? Or is this call supposed to support both?

Reply:
it supports both